src/descent/steepest.jl

```diff
@@ -0,0 +1,66 @@
+"""
+    SteepestDescent(; linsolve = nothing, precs = DEFAULT_PRECS)
+
+Compute the descent direction as ``δu = -Jᵀfu``.
```

Comment:
why is this steepest?

Reply:
That's the gradient of $\frac{1}{2}||f(x)||_2$